### DISCUSSION TITLE
Check ALL passwords periodically to generate reports

### DISCUSSION THREAD

**Discussion by JaneJeon** [dani-garcia/vaultwarden#2177] [Ideas]
This is a missing (paid) feature from the upstream project, but Bitwarden offers [reports](https://bitwarden.com/help/article/reports/) to check the "health" of its passwords, including:
- weak passwords
- pwned passwords
- reused passwords
- unsecured websites (http instead of https)
- data breach report
- etc

Obviously, it doesn't have to be a fancy schmancy reporting UI; it could simply be a weekly email of "data dumps"

**Comment by BlackDex** [comment]
This is also all available for Vaultwarden, just look at the same locations as mentioned in the link you provided.

Also, there is  no way to do this server side, not for Vaultwarden and also not for Bitwarden since it's not possible to decrypt the vault data without a user provide key, in this case the password.
